src/app/dev/DevToys.Tools/Tools/Formatters/Xml/XmlFormatterGuiTool.cs

```diff
@@ -0,0 +1,185 @@
+                            Item(XmlFormatter.TwoSpaces, Indentation.TwoSpaces),
+                            Item(XmlFormatter.FourSpaces, Indentation.FourSpaces),
+                            Item(XmlFormatter.OneTab, Indentation.OneTab),
+                            Item(XmlFormatter.Minified, Indentation.Minified)
+                        ),
+                        Setting("xml-text-newLineOnAttributes-setting")
+                        .Icon("FluentSystemIcons", '\uf7ed')
+                        .Title(XmlFormatter.NewLineOnAttributes)
+                        .Handle(
+                            _settingsProvider,
+                            newLineOnAttributes,
+                            stateDescriptionWhenOn: XmlFormatter.NewLineOnAttributesDescription,
+                            stateDescriptionWhenOff: null,
+                            OnSettingChanged
+                        )
+                    )
+                ),
+                Cell(
+                    GridRow.Content,
+                    GridColumn.Content,
+                    SplitGrid()
+                        .Vertical()
+                        .WithLeftPaneChild(
+                            _inputTextArea
+                                .Title(XmlFormatter.Input)
+                                .OnTextChanged(OnInputTextChanged))
+                        .WithRightPaneChild(
+                            _outputTextArea
+                                .Title(XmlFormatter.Output)
+                                .ReadOnly()
+                                .Extendable())
+                )
+            )
+        );
+
+    // Smart detection handler.
+    public void OnDataReceived(string dataTypeName, object? parsedData)
+    {
+        if (dataTypeName == PredefinedCommonDataTypeNames.Xml &&
+            parsedData is string xmlStrongTypedParsedData)
+        {
+            _inputTextArea.Language(XmlLanguage);
+            _outputTextArea.Language(XmlLanguage);
+            _inputTextArea.Text(xmlStrongTypedParsedData);
+        }
+    }
+
+    public void Dispose()
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _semaphore.Dispose();
+    }
+
+    private void OnSettingChanged(bool value)
+    {
+        StartFormat(_inputTextArea.Text);
+    }
+
+    private void OnIndentationModelChanged(Indentation indentationMode)
+    {
+        StartFormat(_inputTextArea.Text);
+    }
+
+    private void OnInputTextChanged(string text)
+    {
+        StartFormat(text);
+    }
+
+    private void StartFormat(string text)
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _cancellationTokenSource = new CancellationTokenSource();
+
+        WorkTask = FormatAsync(text, _cancellationTokenSource.Token);
+    }
+
+    private async Task FormatAsync(string input, CancellationToken cancellationToken)
```

Comment:
I would suggest to pass `indentationMode` and `newLineOnAttributes` in arguments of this method because like 172 might may the execution flow waiting for quite some time and it might be better to keep the method running with the state of the settings are they were at the moment of when the task has been scheduled.

Reply:
Good idea. Fixed!